llvm/test/Transforms/LowerWGLocalMemoryCall/group_local_memory.ll

```diff
@@ -0,0 +1,286 @@
+entry:
+  %0 = getelementptr inbounds %"class._ZTSN2cl4sycl2idILi1EEE.cl::sycl::id", %"class._ZTSN2cl4sycl2idILi1EEE.cl::sycl::id"* %_arg_3, i64 0, i32 0, i32 0, i64 0
+  %1 = addrspacecast i64* %0 to i64 addrspace(4)*
+  %2 = load i64, i64 addrspace(4)* %1, align 8
+  %add.ptr.i = getelementptr inbounds i32, i32 addrspace(1)* %_arg_, i64 %2
+  %3 = getelementptr inbounds %"class._ZTSN2cl4sycl2idILi1EEE.cl::sycl::id", %"class._ZTSN2cl4sycl2idILi1EEE.cl::sycl::id"* %_arg_8, i64 0, i32 0, i32 0, i64 0
+  %4 = addrspacecast i64* %3 to i64 addrspace(4)*
+  %5 = load i64, i64 addrspace(4)* %4, align 8
+  %add.ptr.i32 = getelementptr inbounds float, float addrspace(1)* %_arg_4, i64 %5
+  %6 = load <3 x i64>, <3 x i64> addrspace(4)* addrspacecast (<3 x i64> addrspace(1)* @__spirv_BuiltInWorkgroupId to <3 x i64> addrspace(4)*), align 32, !noalias !5
+  %7 = extractelement <3 x i64> %6, i64 0
+  %8 = load <3 x i64>, <3 x i64> addrspace(4)* addrspacecast (<3 x i64> addrspace(1)* @__spirv_BuiltInLocalInvocationId to <3 x i64> addrspace(4)*), align 32, !noalias !12
+  %9 = extractelement <3 x i64> %8, i64 0
+  %ptridx.i26.i = getelementptr inbounds i32, i32 addrspace(1)* %add.ptr.i, i64 %7
+  %ptridx.ascast.i27.i = addrspacecast i32 addrspace(1)* %ptridx.i26.i to i32 addrspace(4)*
+  %call.i22.i = tail call spir_func i8 addrspace(3)* @__sycl_allocateLocalMemory(i64 128, i64 4) #2, !noalias !17
+  %shift = shufflevector <3 x i64> %8, <3 x i64> poison, <3 x i32> <i32 undef, i32 2, i32 undef>
+  %10 = or <3 x i64> %shift, %8
+  %11 = extractelement <3 x i64> %10, i64 1
+  %12 = or i64 %11, %9
+  %13 = icmp eq i64 %12, 0
+  br i1 %13, label %if.then.i23.i, label %_ZN2cl4sycl18group_local_memoryI3FooNS0_5groupILi1EEEJiRiEEENSt9enable_ifIXaasr3std25is_trivially_destructibleIT_EE5valuesr6detail8is_groupIT0_EE5valueENS0_9multi_ptrIS7_LNS0_6access13address_spaceE3EEEE4typeES8_DpOT1_.exit.i
+
+if.then.i23.i:                                    ; preds = %entry
+  %14 = bitcast i8 addrspace(3)* %call.i22.i to i32 addrspace(3)*
+  ; CHECK-COUNT-33: i8 addrspace(3)* getelementptr inbounds ([128 x i8], [128 x i8] addrspace(3)* [[WGLOCALMEM_1]], i32 0, i32 0)
```

Comment:
Can we reduce the number of instructions? What is the reason to have 33 similar instructions?

Reply:
Done